commit 4ed8f337dee32df71435689c19d22e4ee846e15a
Author: Masami Hiramatsu (Google) <mhiramat@kernel.org>
Date:   Tue Jul 11 23:15:57 2023 +0900

    Revert "tracing: Add "(fault)" name injection to kernel probes"
    
    This reverts commit 2e9906f84fc7c99388bb7123ade167250d50f1c0.
    
    It was turned out that commit 2e9906f84fc7 ("tracing: Add "(fault)"
    name injection to kernel probes") did not work correctly and probe
    events still show just '(fault)' (instead of '"(fault)"'). Also,
    current '(fault)' is more explicit that it faulted.
    
    This also moves FAULT_STRING macro to trace.h so that synthetic
    event can keep using it, and uses it in trace_probe.c too.
    
    Link: https://lore.kernel.org/all/168908495772.123124.1250788051922100079.stgit@devnote2/
    Link: https://lore.kernel.org/all/20230706230642.3793a593@rorschach.local.home/
    
    Cc: stable@vger.kernel.org
    Cc: Andrew Morton <akpm@linux-foundation.org>
    Cc: Tom Zanussi <zanussi@kernel.org>
    Signed-off-by: Masami Hiramatsu (Google) <mhiramat@kernel.org>
    Reviewed-by: Steven Rostedt (Google) <rostedt@goodmis.org>

diff --git a/kernel/trace/trace_probe.c b/kernel/trace/trace_probe.c
index 2d2616678295..591399ddcee5 100644
--- a/kernel/trace/trace_probe.c
+++ b/kernel/trace/trace_probe.c
@@ -65,7 +65,7 @@ int PRINT_TYPE_FUNC_NAME(string)(struct trace_seq *s, void *data, void *ent)
 	int len = *(u32 *)data >> 16;
 
 	if (!len)
-		trace_seq_puts(s, "(fault)");
+		trace_seq_puts(s, FAULT_STRING);
 	else
 		trace_seq_printf(s, "\"%s\"",
 				 (const char *)get_loc_data(data, ent));